tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -416,7 +416,10 @@ Long64_t TTreeIndex::FindValues(Long64_t major, Long64_t minor) const
 Long64_t TTreeIndex::GetEntryNumberWithBestIndex(Long64_t major, Long64_t minor) const
 {
    if (fN == 0) return -1;
-
+   if ((minor > 0x7fffffff && major != 0) || major > 0x1ffffffff) {
```

Comment:
So `0x7fffffff` is `std::numeric_limits<int>::max()` and it should be written that way. I don't know what `0x1ffffffff` corresponds to. In any case, both these values are within the range of `Long64_t`, so can we write them as decimal integers and explain why the condition is doing the checks that it is doing?

Reply:
major starts at bit 31, so it has maximum 33 bits of size to fit in  64bit, so 0x1fff... is pow(2,33)-1

I can write them as decimals, but I thought it was better to be coherent with the hex formulation within: https://github.com/root-project/root/blob/81172612fa1caee4d01d23ce18e35eae90fd36ea/tree/treeplayer/src/TTreeIndex.cxx#L326